### PR TITLE
Update operator to 1.125.0

### PR DESCRIPTION
**Pull request by yuwenma**

### CODE REVIEW

File: operator/config/gke-addon/image_configmap.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-  cnrm.controller: gcr.io/gke-release/cnrm/controller:7a86865
-  cnrm.deletiondefender: gcr.io/gke-release/cnrm/deletiondefender:7a86865
-  cnrm.recorder: gcr.io/gke-release/cnrm/recorder:7a86865
-  cnrm.unmanageddetector: gcr.io/gke-release/cnrm/unmanageddetector:7a86865
-  cnrm.webhook: gcr.io/gke-release/cnrm/webhook:7a86865
+  cnrm.controller: gcr.io/gke-release/cnrm/cnrm/controller:2fa0f72
+  cnrm.deletiondefender: gcr.io/gke-release/cnrm/cnrm/deletiondefender:2fa0f72
+  cnrm.recorder: gcr.io/gke-release/cnrm/cnrm/recorder:2fa0f72
+  cnrm.unmanageddetector: gcr.io/gke-release/cnrm/cnrm/unmanageddetector:2fa0f72
+  cnrm.webhook: gcr.io/gke-release/cnrm/cnrm/webhook:2fa0f72
```
Comment (yuwenma) [inline operator/config/gke-addon/image_configmap.yaml:7]: This is skeptical, may due to recent changes to some release scripts. But I don't think it should block the 1.125 release. I'll investigate offline

Comment (jingyih) [reply operator/config/gke-addon/image_configmap.yaml:7]: I checked the first one. The image exists for `gcr.io/gke-release/cnrm/controller:2fa0f72`. There is no image for `gcr.io/gke-release/cnrm/cnrm/controller:2fa0f72`.

Comment (jingyih) [reply operator/config/gke-addon/image_configmap.yaml:7]: Could we manually update the image path for these?

I believe the plan is to push for a Config Connector version upgrade in the GKE addon to this version, as it resolves the GKE node pool issue which is fixed in this release.

/cc @cheftako

**Review by yuwenma — COMMENTED** [review]
/hold

File: operator/autopilot-channels/packages/configconnector/1.125.0/crds.yaml [hunk 2]
```diff
@@ -45782,18 +47500,12 @@ spec:
     storage: true
     subresources:
       status: {}
-status:
```
Comment (cheftako) [inline operator/autopilot-channels/packages/configconnector/1.125.0/crds.yaml:45785]: Why remove this block?

File: operator/config/gke-addon/image_configmap.yaml (hunk 1 above)
Comment (cheftako) [reply operator/config/gke-addon/image_configmap.yaml:7]: Good catch. We should be generating these images at the new paths. Can we have @xiaoweim look at why the image(s) was not generated?

File: operator/autopilot-channels/packages/configconnector/1.125.0/crds.yaml (hunk 2 above)
Comment (yuwenma) [reply operator/autopilot-channels/packages/configconnector/1.125.0/crds.yaml:45785]: This is a good change from controller-gen. We don't want to deploy object (CRDs or other kind) to the cluster with a status.

File: operator/config/gke-addon/image_configmap.yaml (hunk 1 above)
Comment (yuwenma) [reply operator/config/gke-addon/image_configmap.yaml:7]: Searched around, this is the only change we had around the image in the past month https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2667#discussion_r1773743059
Though I don't see how that relates to this change. @xiaoweim This currently blocks 1.125 release. Could you investigate?

Comment (yuwenma) [reply operator/config/gke-addon/image_configmap.yaml:7]: cc @justinsb just FYI.

File: operator/autopilot-channels/packages/configconnector/1.125.0/cluster/gcp-identity/0-cnrm-system.yaml [hunk 3]
```diff
@@ -2693,15 +2753,15 @@ spec:
...
+        cnrm.cloud.google.com/version: 1.125.0
       labels:
         cnrm.cloud.google.com/component: cnrm-deletiondefender
         cnrm.cloud.google.com/system: "true"
     spec:
       containers:
       - command:
         - /configconnector/deletiondefender
-        image: gcr.io/gke-release/cnrm/deletiondefender:e373e18
+        image: gcr.io/gke-release/cnrm/cnrm/deletiondefender:2fa0f72
```
[the remark this answers is not in the document]
Comment (yuwenma) [reply operator/autopilot-channels/packages/configconnector/1.125.0/cluster/gcp-identity/0-cnrm-system.yaml]: Could you share which manifest you refer to?

File: operator/config/gke-addon/image_configmap.yaml (hunk 1 above)
Comment (cheftako) [reply operator/config/gke-addon/image_configmap.yaml]: Sorry I should have said why we have a redundant "cnrm/cnrm"

**Review by jingyih — COMMENTED** [review]
/lgtm

File: operator/autopilot-channels/packages/configconnector/1.125.0/crds.yaml [hunk 4]
```diff
@@ -10122,24 +10144,21 @@ apiVersion: apiextensions.k8s.io/v1
...
   group: bigqueryanalyticshub.cnrm.cloud.google.com
   names:
     categories:
     - gcp
     kind: BigQueryAnalyticsHubDataExchange
+    listKind: BigQueryAnalyticsHubDataExchangeList
     plural: bigqueryanalyticshubdataexchanges
-    shortNames:
-    - gcpbigqueryanalyticshubdataexchange
-    - gcpbigqueryanalyticshubdataexchanges
```
Comment (acpana) [inline operator/autopilot-channels/packages/configconnector/1.125.0/crds.yaml:10142]: Addressing in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3229

**Pull request merged**